classes/01class/exercises/c01-aws02/gpavelar/README.md

```diff
@@ -0,0 +1,203 @@
+# C01-AWS02
+
+## Commands Execution Output
+
+- Commands to create the S3 Bucket and to copy the file to the bucket:
+
+```bash
+## Creating bucket S3 - first way
+$ aws s3 mb s3://mydevops-bucket
+# output
+make_bucket: mydevops-bucket
+
+## Creating bucket S3 - second way
+$ aws s3api create-bucket --bucket myseconddevops-bucket --region ap-southeast-2
+# output
+/myseconddevops-bucket
+
+## Listing bucket S3
+$ aws s3 ls
+# output
+2020-06-20 13:00:33 mydevops-bucket
+2020-06-20 13:07:04 myseconddevops-bucket
+## or
+$ aws s3api list-buckets
+# output
+BUCKETS 2020-06-20T03:00:33+00:00       mydevops-bucket
+BUCKETS 2020-06-20T03:07:04+00:00       myseconddevops-bucket
+OWNER   gpavelardev     ***********
+
+## copying a file to my s3 bucket.
+aws s3 cp devops-classes/devops-rules.txt s3://mydevops-bucket/devops-classes
+## Another test
+aws s3 cp devops-classes/devops-aws.txt s3://mydevops-bucket/devops-classes/ --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers
+```
+
+- Commands to allow the EC2 instance to access the files in S3:
+
+```bash
+## Acessing instance via SSH. I stopped this instance
+$ ssh -A ec2-user@3.104.223.187
+## Trying to list aws s3
+$ aws s3 ls
+# output
+Unable to locate credentials. You can configure credentials by running "aws configure".
+$ aws configure
+## Added my access key id and secret access key, region, etc.
+
+## Creating role
+aws iam create-role --role-name Test-Role --assume-role-policy-document file://Role-Read-Trust-Policy.json
+# output
+
+ROLE arn:aws:iam::478433196210:role/Test-Role 2020-06-21T13:44:23+00:00 / AROAW6ZGXHCZNG4L6UYON Test-Role
+ASSUMEROLEPOLICYDOCUMENT 2012-10-17
+STATEMENT sts:AssumeRole Allow
+PRINCIPAL ec2.amazonaws.com
+```
+
+`Role-Read-Trust-Policy.json` file used:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": {
+    "Effect": "Allow",
+    "Principal": { "Service": "ec2.amazonaws.com" },
+    "Action": "sts:AssumeRole"
+  }
+}
+```
+
+```bash
+
+## Creating my own policy
+aws iam create-policy --policy-name AmazonS3ReadOnlyAccess_test \
+--policy-document  file://my-own-s3-policy.json
+
+# output
+POLICY  arn:aws:iam::478433196210:policy/AmazonS3ReadOnlyAccess_test    0       2020-07-02T04:35:52+00:00       v1      True    /       0       ANPAW6ZGXHCZAWLMGHLPC   AmazonS3ReadOnlyAccess_test     2020-07-02T04:35:52+00:00
+
+
+## Attch a managed policy to an IAM role
+aws iam attach-role-policy --role-name Test-Role --policy-arn arn:aws:iam::478433196210:policy/AmazonS3ReadOnlyAccess_test
+# output
+NONE
+
+```
+
+`my-own-s3-policy.json`
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "VisualEditor0",
+      "Effect": "Allow",
+      "Action": ["s3:GetObject", "s3:ListBucket"],
+      "Resource": ["arn:aws:s3:::mydevops-bucket", "arn:aws:s3:::mydevops-bucket/*"]
+    }
+  ]
+}
+```
+
+Continuando as alterações e removendo uma policy do IAM ROLE
```

Comment:
English please :)

Reply:
OMG,sorry. I don't know why I wrote in portuguese.